pkg/vmprovider/providers/vsphere/constants/constants.go

```diff
@@ -128,4 +128,7 @@ const (
 	// BackupVMDiskDataExtraConfigKey is the ExtraConfig key to the VM's disk info
 	// data in JSON, compressed using gzip and base64-encoded.
 	BackupVMDiskDataExtraConfigKey = "vmservice.virtualmachine.diskdata"
+	// BackupVMInstanceIDExtraConfigKey is the ExtraConfig key to the VM's
+	// Cloud-Init instance ID, compressed using gzip and base64-encoded.
+	BackupVMInstanceIDExtraConfigKey = "vmservice.virtualmachine.instanceid"
```

Comment:
I think we should move these to the `api/` package to make these part of the API contract.  Once we start shipping these, we need to support these for backup/restore to work.

Reply:
Hmm I put them under this file because there are already other ExtraConfig keys declared here. I thought it would be better to have them all in one place for easier lookup.